File: 1910. Remove All Occurrences of a Substring.rb
#* 1910. Remove All Occurrences of a Substring
s = "daabcbaabcbc"; part = "abc"
# @param {String} s
# @param {String} part
# @return {String}
#* 1
def remove_occurrences(s, part)
  n = part.size
  i = 0

  while i <= s.size - n
    if s[i, n] == part
      s.slice!(i, n)
      #reset index to check for overlapping parts
      i = [0, i - n].max
    else
      i += 1
    end
  end
  s
end
p remove_occurrences(s, part)

#* WORKING . . .
def remove_occurrences(s, part)
  n=part.size
    (0..s.size-n).each do |i|
      s.slice!(i...n+i) if s[i...n+i] == part
    end
    s
end
s = "daabcbaabcbc"
part = "abc"
p remove_occurrences(s, part)

File: 1910. Remove All Occurrences of a Substring.rs
// 1910. Remove All Occurrences of a Substring
//s = "daabcbaabcbc"; part = "abc"
struct Solution;

impl Solution {
    pub fn remove_occurrences(mut s: String, part: String) -> String {
        while let Some(pos) = s.find(&part) {
            s.drain(pos..pos + part.len());
        }
        s
    }
}

fn main() {
    let s = "daabcbaabcbc".to_string();
    let part = "abc".to_string();
    println!("{}", Solution::remove_occurrences(s, part)); // Outputs "dababcbc"
}
